ext/openssl/openssl.c

```diff
@@ -7642,14 +7642,14 @@ PHP_FUNCTION(openssl_cipher_iv_length)
 PHP_OPENSSL_API zend_string* php_openssl_random_pseudo_bytes(zend_long buffer_length)
 {
 	zend_string *buffer = NULL;
-	if (buffer_length <= 0
-#ifndef PHP_WIN32
-		|| ZEND_LONG_INT_OVFL(buffer_length)
-#endif
-			) {
+	if (buffer_length <= 0) {
 		zend_argument_value_error(1, "must be greater than 0");
 		return NULL;
 	}
+	if (ZEND_LONG_INT_OVFL(buffer_length)) {
+		zend_argument_value_error(1, "must be less than 2147483648");
```

Comment:
is it possible to write `INT_MAX` instead ?

Reply:
Possible, but super ugly, since `INT_MAX` evaluates to an int, not a string, so we needed to do something like:
````
#define STR_INDIR(x) #x
#define STR(x) STR_INDIR(x)
		zend_argument_value_error(1, "must be less than or equal to " STR(INT_MAX));
````

Reply:
well I meant its value :)

Reply:
nevermind otherwise it s just a detail :)

Reply:
Ah, I see! I think we can leave it as is; after all, it is unlikely that anybody gets that error message (who would even think about getting 2MB of random bytes).

Reply:
I think you could do something like

```suggestion
		zend_argument_value_error(1, "must be less than %u", (unsigned)INT_MAX + 1);
```

but it requires an unsigned cast to avoid a warning (UB?) when compiling on 32-bit systems (even though it will never be executed), or simpler

```suggestion
		zend_argument_value_error(1, "must be less than or equal to %d", INT_MAX);
```

?

Reply:
I think @cmb69 got the idea most likely :)

Reply:
Oh, right, that's a format style function. [I've made it so](https://github.com/php/php-src/commit/5d62cfbc7d4bad5c22818eedc5f249abb62abdc2) :)